src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4845,7 +4845,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           case Select(qualqual, vname) =>
             gen.evalOnce(qualqual, context.owner, context.unit) { qq =>
               val qq1 = qq()
-              mkAssign(Select(qq1, vname) setPos qual.pos)
+              mkAssign(Select(qq1, qual.symbol) setPos qual.pos)
```

Comment:
While in the neighbourhood, there might be some unpositioned synthetic trees created just above (the `Select` in `mkCall` and the `Apply` in `mkApplies`.

Reply:
Nice catch. I actually nabbed these over in #6423 (028510775be278f9204ed5d5f9e1dc2616771e7e).